### PR TITLE
lib/storage/partition: add check to ensure parts exist on disk

### PR DESCRIPTION
**Pull request by zekker6**
If part exists in parts.json but is missing on disk there will be a misleading error similar to "unexpected number of substrings in the part name".

This change forces verification of part existence and throws a correct error in case it is missing on disk.

Such issue can be result of https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5005 or disk corruption.

### CODE REVIEW

**Comment by hagen1778** [comment]
Builds shouldn't fail after https://github.com/VictoriaMetrics/VictoriaMetrics/commit/7b99781b09e6fb6884c50dc955622db2a8111276

File: lib/storage/partition.go [hunk 1]
```diff
@@ -1779,6 +1779,11 @@ func mustOpenParts(path string, partNames []string) []*partWrapper {
 	des := fs.MustReadDir(path)
 	m := make(map[string]struct{}, len(partNames))
 	for _, partName := range partNames {
+		partPath := filepath.Join(path, partName)
+		if !fs.IsPathExist(partPath) {
+			logger.Panicf("FATAL: missing part dir %q; part is listed in %q, but missing on disk", partPath, path+partsFilename)
```
Comment (hagen1778) [inline lib/storage/partition.go:1784]: Should we add an action list to the error message? Like: "please check the backup was restored completely", or "verify there are no disk corruptions and restore from the backup", or "delete this part from parts.json if nothing helped"?

Comment (zekker6) [reply lib/storage/partition.go]: Updated error message, ptal

> "please check the backup was restored completely"

this case is checked upon storage start and triggers respective error :)

**Review by hagen1778 — APPROVED** [review]

**Comment by valyala** [comment]
@zekker6 , thanks for the pull request!

**Comment by valyala** [comment]
FYI, this pull request has been included in [v1.94.0](https://docs.victoriametrics.com/CHANGELOG.html).

**Pull request merged**